src/main/java/com/training/web/pages/flipkart/ResultPage.java

```diff
@@ -59,4 +63,25 @@ public ResultPage clickNextPage() {
         Collections.sort(sortedPrice);
         return sortedPrice;
     }
+
+    public ResultPage clickProduct(int itemNumber) throws InterruptedException {
+        itemNumber=itemNumber-1;
+        productList.get(itemNumber).click();
+        return this;
+    }
+
+    public ResultPage getProductsList() throws InterruptedException {
```

Comment:
I think its better if you return this as a list, rather than defining a global variable